azure-pipelines.yml

```diff
@@ -2,72 +2,107 @@ trigger:
+    - task: GitHubRelease@1
+      condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags/'))
+      inputs:
+        gitHubConnection: 'github.com_marianan'
+        repositoryName: '$(Build.Repository.Name)'
+        action: 'create'
+        target: '$(Build.SourceVersion)'
+        tagSource: 'userSpecifiedTag'
+        tag: '$(Build.SourceBranch)'
+        releaseNotesSource: 'inline'
+        assets: '$(Build.ArtifactStagingDirectory)/Package/*'
+        isPreRelease: true
+        addChangeLog: false
 
-  - task: CopyFiles@2
-    inputs:
-      SourceFolder: '$(Build.SourcesDirectory)'
-      Contents: '*.tgz'
-      TargetFolder: '$(Build.ArtifactStagingDirectory)/Package'
-  - task: PublishBuildArtifacts@1
+- job: UploadToStorage
+  dependsOn: Build
+  condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags/'))
+  pool:
+    name: Azure-IoT-EdgeExperience-1ES-Hosted-Windows
+    demands:
+      - ImageOverride -equals MMS2022TLS
+  
+  steps:
+  - task: Bash@3
     inputs:
-      PathtoPublish: '$(Build.ArtifactStagingDirectory)/Package'
-      ArtifactName: 'drop'
-      publishLocation: 'Container'
+      targetType: 'inline'
+      script: |
+        if [[ $(Build.SourceBranch) =~ ^refs/tags/v?[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+        echo "##vso[task.setvariable variable=NPM_PUBLISH_FLAG;]true"
+        fi
 
-  - task: Npm@1
-    displayName: "npm Test"
+  - task: DownloadPipelineArtifact@2
     inputs:
-      command: 'custom'
-      customCommand: 'test'
-    
-  - task: GitHubRelease@1
-    condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags/'))
+      buildType: 'current'
+      artifactName: 'drop'
+      targetPath: '$(Build.ArtifactStagingDirectory)/Package'
+
+  - task: Bash@3
     inputs:
-      gitHubConnection: 'github.com_marianan'
-      repositoryName: '$(Build.Repository.Name)'
-      action: 'create'
-      target: '$(Build.SourceVersion)'
-      tagSource: 'userSpecifiedTag'
-      tag: '$(Build.SourceBranch)'
-      releaseNotesSource: 'inline'
-      assets: '$(Build.ArtifactStagingDirectory)/Package/*'
-      addChangeLog: false
+      targetType: 'inline'
+      script: |
+        VERSION=$(node -p -e "require('./package.json').version")
+        echo "##vso[task.setvariable variable=PACKAGEVERSION;]$VERSION"
 
-  - task: Npm@1
-    displayName: "npm Publishing"
-    condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags/'))
+  - task: AzureFileCopy@3
+    condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags/'), eq(variables['NPM_PUBLISH_FLAG'], 'true'))
     inputs:
-      command: 'publish'
-      publishRegistry: 'useFeed'
-      publishFeed: '"Service connection name"'      
\ No newline at end of file
+      SourcePath: '$(System.DefaultWorkingDirectory)/artifacts/*'
+      azureSubscription: 'azuresdkpartnerdrops'
```

Comment:
>azuresdkpartnerdrops [](http://example.com/codeflow?start=26&length=20)

could you give me edit access to the service connection?

Reply:
done